test/module/shared_model/interface/common_objects/amount_test.cpp

```diff
@@ -10,43 +10,40 @@
 #include <utility>
 
 #include <gtest/gtest.h>
-#include <boost/multiprecision/cpp_int.hpp>
 
 using namespace shared_model::interface;
 
 struct AmountTest : public ::testing::Test {
-  using int_type = std::decay_t<decltype(std::declval<Amount>().intValue())>;
-
-  /// Check int value, precision and string representation of a valid amount.
+  /// Check sign, precision and string representation of a valid amount.
   void checkValid(const Amount &tested,
-                  int_type ref_int_val,
+                  bool greater_than_zero,
```

Comment:
let's conform to the interface

```suggestion
                  int ref_sign,
```

Reply:
>Returns a value less than zero if *this is negative, a value greater than zero if *this is positive, and zero if *this is zero. 

Which value should be passed for comparison?

Reply:
Agreed on using `boost::math::sign` function to test `Amount::sign()` result.